test-harness/index.js

```diff
@@ -0,0 +1,270 @@
+        describe('Dynamic response', () => {
+          test('should default to 200 and mock from schema', async () => {
+            const { reqRes, masterFile } = await runTest(requests[11]);
+            const payload = reqRes.response.body;
+
+            expect(reqRes.response.status).toBe(masterFile.response.status);
+            expect(reqRes.response.status).toBe(200);
+
+            expect(payload).toHaveProperty('id');
+            expect(payload).toHaveProperty('username');
+            expect(payload).toHaveProperty('firstName');
+            expect(payload).toHaveProperty('lastName');
+            expect(payload).toHaveProperty('email');
+            expect(payload).toHaveProperty('password');
+            expect(payload).toHaveProperty('phone');
+            expect(payload).toHaveProperty('userStatus');
+          });
+        });
+      });
+
+      describe('When sending a GET with all the needed parameters in query', () => {
+        describe('Static response', () => {
+          test('A response with status code 200 is returned', async () => {
+            const { reqRes, masterFile } = await runTest(requests[1]);
+
+            expect(reqRes).toStrictEqual(masterFile);
+          });
+        });
+      });
+
+      xdescribe('Authorization', () => {
+        xdescribe('When a request does not include authorization details', () => {
+          test(['Should get 401 for not being authorized', 'NOT_IN_V3:SO-176'].join(), async () => {
+            const { reqRes, masterFile } = await runTest(requests[3]);
+
+            expect(reqRes.response.status).toBe(401);
+            // expect(reqRes.response.status).toBe(masterFile.response.status);
+          });
+        });
+      });
+
+      describe('Dynamic paths', () => {
+        describe('When a resource is request from a dynamic path like /pets/{petId}', () => {
+          describe('Dynamic response', () => {
+            test('The resource is sent back with a proper schema', async () => {
+              const { reqRes, masterFile } = await runTest(requests[4]);
+              const payload = reqRes.response.body;
+
+              expect(reqRes.response.status).toBe(masterFile.response.status);
+
+              expect(payload.name).toBeDefined();
+              expect(payload.photoUrls).toBeDefined();
+              expect(payload.id).toBeDefined();
+              expect(payload.category).toBeDefined();
+              expect(payload.tags).toBeDefined();
+              expect(payload.status).toBeDefined();
+            });
+          });
+        });
+      });
+
+      xdescribe('When a form data is sent with application/x-www-form-urlencoded to create a single resource', () => {
+        test(
+          [
+            'Should answer with JSON representing the resource',
+            'prism v3 does not respect x-ww-form-urlencoded',
+          ].join(),
+          async () => {
+            const { reqRes, masterFile } = await runTest(requests[5]);
```

Comment:
Maybe use [each](https://jestjs.io/docs/en/api#testeachtable-name-fn-timeout) instead of looping through all the tests manually?

Reply:
In cases when we do not assert `expect(reqRes).toStrictEqual(masterFile);`, where do I put assertions then? Since I cannot just `expect(reqRes).toStrictEqual(masterFile);` for a dynamic response, how do I now change this? Do you think all of the assertions should be moved to [`requests.js`](`https://github.com/stoplightio/prism/blob/7dab5507392b1108b7e4438177e5c9b6ab5403af/test-harness/requests.js`)? And also, where do I move `describe`s and `test`s content (descriptive strings)?